### PR TITLE
Change Camera orientation to FLU

### PR DESCRIPTION
**Pull request by lukicdarkoo**
Similar to #3493, this PR convert the camera to FLU

**Documentation**:
https://cyberbotics.com/doc/reference/camera?version=feature-enu-camera

**TODO**:
- [x] update all tests
- [x] fix usage of camera subnodes (e.g. focus)
- [x] fix docs

### CODE REVIEW

File: projects/samples/devices/worlds/camera.wbt
Comment (omichel) [inline projects/samples/devices/worlds/camera.wbt:268]: That looks odd, can't we simplify?
```suggestion
      translation 0 0.0915 -0.04
      rotation 1 0 0 3.14159
```

File: projects/samples/devices/worlds/camera.wbt
Comment (omichel) [inline projects/samples/devices/worlds/camera.wbt:271]:
```suggestion
          translation 0 0 -0.015
```

File: src/webots/nodes/WbCamera.cpp
Comment (omichel) [inline src/webots/nodes/WbCamera.cpp:1014]:
```suggestion
  if (hasBeenSetup())
    mWrenCamera->rotatePitch(M_PI);  // OpenCV camera axis orientation
```

File: tests/api/worlds/camera_revert.wbt
Comment (omichel) [inline tests/api/worlds/camera_revert.wbt:217]:
```suggestion
          translation 0 0 -0.015
```

File: tests/api/worlds/robot_multiple_step.wbt
Comment (omichel) [inline tests/api/worlds/robot_multiple_step.wbt:54]:
```suggestion
      rotation 0.70710528118436 0 0.707108281185553 3.14159
```

File: projects/samples/devices/worlds/camera_auto_focus.wbt
Comment (omichel) [inline projects/samples/devices/worlds/camera_auto_focus.wbt:171]: Why do we have to change this?

**Review by omichel — CHANGES_REQUESTED** [review]

File: projects/samples/devices/worlds/camera_auto_focus.wbt
Comment (lukicdarkoo) [reply projects/samples/devices/worlds/camera_auto_focus.wbt]: Slipped :/

**Comment by lukicdarkoo** [comment]
I want to compare with Gazebo before merging. Because, as stated here:

> In the case of cameras, there is often a second frame defined with a "_optical" suffix. This uses a slightly different convention:
> - z forward
> - x right
> - y down

there are apparently two frames, one FLU and the other the OpenCV one.

**Comment by lukicdarkoo** [comment]
A good example is [realsense_ros_gazebo](https://github.com/nilseuropa/realsense_ros_gazebo).

In the example, the camera device is associated with `${sensor_name}_link` which follows the FLU convention. The transform tree continues as follows:
```
${sensor_name}_link (FLU)
    |-- ${sensor_name}_depth_frame (FLU)
        |-- ${sensor_name}_depth_optical_frame (OpenCV)
```

Here is a RViz snapshot:
image

Therefore, they have a dummy link `${sensor_name}_depth_optical_frame` to get the optical frame. I am not sure how does it reflect in our models. To me, the Camera node should be FLU and a user can add a child node to get the optical frame.

**Comment by lukicdarkoo** [comment]
Considering that all camera devices should follow the FLU convention, do you think it is a good idea to convert the WREN camera? 
I have already started in the `feature-flu-camera` branch, but many things are broken. I am not sure what else we will discover.

**Comment by stefaniapedrazzi** [comment]
> Considering that all camera devices should follow the FLU convention, do you think it is a good idea to convert the WREN camera?
> I have already started in the `feature-flu-camera` branch, but many things are broken. I am not sure what else we will discover.

Yes, it seem good to convert the WREN camera.
Note that WREN camera is used for all the rendering devices (main viewpoint, camera, range finder, lidar) and not just for camera devices so if they do not use the same convention it might be needed to adjust the orientation in some cases.
But the main viewpoint and camera devices are probably the most used functionalities.

**Comment by omichel** [comment]
I assume this PR is on hold until #3525 is merged.

**Comment by lukicdarkoo** [comment]
> I assume this PR is on hold until #3525 is merged.

It should be replaced by #3525 (along with conversion of all other camera based devices)

**Comment by omichel** [comment]
It looks all good to me, but I would try to improve a little bit the documentation image to make it easier to understand, maybe with a narrower angle camera and without selecting the robot, but only the camera.

**Comment by lukicdarkoo** [comment]
> It looks all good to me, but I would try to improve a little bit the documentation image to make it easier to understand, maybe with a narrower angle camera and without selecting the robot, but only the camera.

Good point, I have just replaced the image. If you have any other on how to improve the image let me know. I believe those images are very important.

**Review by omichel — APPROVED** [review]
It is way better. Thank you.

**Pull request merged**
